pkg/store/matchers.go

```diff
@@ -0,0 +1,71 @@
+package store
+
+import (
+	"github.com/improbable-eng/thanos/pkg/store/storepb"
+	"github.com/prometheus/tsdb/labels"
+	"github.com/pkg/errors"
+	"bytes"
+	"strconv"
+)
+
+func translateMatcher(m storepb.LabelMatcher) (labels.Matcher, error) {
+	switch m.Type {
+	case storepb.LabelMatcher_EQ:
+		return labels.NewEqualMatcher(m.Name, m.Value), nil
+
+	case storepb.LabelMatcher_NEQ:
+		return labels.Not(labels.NewEqualMatcher(m.Name, m.Value)), nil
+
+	case storepb.LabelMatcher_RE:
+		return labels.NewRegexpMatcher(m.Name, m.Value)
+
+	case storepb.LabelMatcher_NRE:
+		m, err := labels.NewRegexpMatcher(m.Name, m.Value)
+		if err != nil {
+			return nil, err
+		}
+		return labels.Not(m), nil
+	}
+	return nil, errors.Errorf("unknown label matcher type %d", m.Type)
+}
+
+func translateMatchers(ms []storepb.LabelMatcher) (res []labels.Matcher, err error) {
+	for _, m := range ms {
+		r, err := translateMatcher(m)
+		if err != nil {
+			return nil, err
+		}
+		res = append(res, r)
+	}
+	return res, nil
+}
+
+func matcherToSelectorQuery(ms []storepb.LabelMatcher) (string, error) {
+	var b bytes.Buffer
+	b.WriteByte('{')
+
+	for i, m := range ms {
+		if i != 0 {
+			b.WriteByte(',')
+		}
+		b.WriteString(m.Name)
+
+		switch m.Type {
+		case storepb.LabelMatcher_EQ:
+			b.WriteByte('=')
+		case storepb.LabelMatcher_NEQ:
+			b.WriteString("!=")
+		case storepb.LabelMatcher_RE:
+			b.WriteString("=~")
+		case storepb.LabelMatcher_NRE:
+			b.WriteString("!~")
+		default:
+			return "", errors.New("unknown matcher type")
+		}
+
+		b.WriteString(strconv.Quote(m.Value))
+	}
+
+	b.WriteByte('}')
+	return b.String(), nil
+}
```

Comment:
Missing newline, shouldn't gofmt fix this automatically?

Reply:
Nope we need to add this to pre commit hook or something

Reply:
I mean, yes - it fixes but I forgot to run it on this file ):

Reply:
It should ideally be a save hook in your editor. `go fmt` is explicitly not intended as a pre-commit hook as I learned recently: https://tip.golang.org/doc/go1.10#gofmt

```
Note that these kinds of minor updates to gofmt are expected from time to time. In general, we recommend against building systems that check that source code matches the output of a specific version of gofmt. For example, a continuous integration test that fails if any code already checked into a repository is not “properly formatted” is inherently fragile and not recommended.
```

At least I interpret that if checking for formatting is discouraged, applying it automatically is so as well.

Reply:
Good to know (: I have that under the special key, but not with the file save